### PR TITLE
Qt4 Cleanup

### PR DESCRIPTION
**Pull request by uklotzde**
A set of trivial changes that delete all obsolete Qt4 code.

The commits have been splitted for easy review:
-  Explicit checks for any Qt 4.x version
- Checks for Qt version < 5.0, i.e. effectively also for any 4.x version
- Checks for Qt version >= 5.0, i.e. the inverse of the previous checks

Emerging merge conflicts should be trivial to resolve, e.g. #2088.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -107,9 +107,6 @@ int main(int argc, char * argv[]) {
     // Support utf-8 for all translation strings. Not supported in Qt 5.
```
Comment (daschuer) [inline src/main.cpp]: I think this is a left over comment.

File: src/preferences/dialog/dlgprefinterface.cpp [hunk 2]
```diff
@@ -110,42 +110,9 @@ DlgPrefInterface::DlgPrefInterface(QWidget * parent, MixxxMainWindow * mixxx,
     connect(ComboBoxSkinconf, SIGNAL(activated(int)), this, SLOT(slotSetSkin(int)));
     connect(ComboBoxSchemeconf, SIGNAL(activated(int)), this, SLOT(slotSetScheme(int)));
 
-
-
-#if QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
```
Comment (daschuer) [inline src/preferences/dialog/dlgprefinterface.cpp]: I think we can reuse parts of that later ... mmm what does this mean here.

File: src/waveform/vsyncthread.cpp [hunk 3]
```diff
@@ -124,17 +116,8 @@ void VSyncThread::swapGl(QGLWidget* glw, int index) {
...
-#else
-#if QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
-#ifdef QT_OPENGL_ES_2
-    glw->swapBuffers();
-#else
-    const QX11Info *xinfo = qt_x11Info(glw);
-    glXSwapBuffers(xinfo->display(), glw->winId());
-#endif
 #else
     glw->swapBuffers();
```
Comment (daschuer) [inline src/waveform/vsyncthread.cpp]: This melds down to one glw->swapBuffers();
It is also part of https://github.com/mixxxdj/mixxx/pull/2088 waiting for merge. 
Can we merge this merge it first and revert this here?

File: src/waveform/vsyncthread.h [hunk 4]
```diff
@@ -7,25 +7,6 @@
 #include <QPair>
 #include <QGLWidget>
 
-#if defined(__APPLE__)
```
Comment (daschuer) [inline src/waveform/vsyncthread.h]: the comment above applies also here.

File: src/waveform/waveformwidgetfactory.cpp [hunk 5]
```diff
@@ -163,8 +163,6 @@ WaveformWidgetFactory::WaveformWidgetFactory() :
             majorVersion = 4;
             minorVersion = 1;
 #endif
-// Flags introduced in Qt 4.7.
-#if QT_VERSION >= QT_VERSION_CHECK(4, 7, 0)
```
Comment (daschuer) [inline src/waveform/waveformwidgetfactory.cpp]: conflicting with #2088

**Review by daschuer — CHANGES_REQUESTED** [review]
Thank you, I have added some comment. This this is for master and #2088 for 2.2.1 IMHO we should merge the later first. 
I don't mind to merge this to 2.2.1 as well by the way.

**Comment by uklotzde** [comment]
I will rebase this PR after merging #2088

**Review by daschuer — APPROVED** [review]

**Comment by daschuer** [comment]
Thank you for taking care. LGTM

**Pull request merged**